File: AndroidXPatternsApp/jmvp-app/src/main/java/com/woowrale/jmvp/di/modules/ServicesModule.java
package com.woowrale.jmvp.di.modules;

import com.woowrale.jmvp.data.repository.ApiClient;
import com.woowrale.jmvp.data.repository.ApiService;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class ServicesModule {

    @Provides
    @Singleton
    ApiService getApiService(){
        return new ApiClient().getClient().create(ApiService.class);
    }
}

File: AndroidPatternsApp/kmvp-app/src/main/java/com/woowrale/kmvp/di/modules/ServicesModule.kt
package com.woowrale.kmvp.di.modules

import com.woowrale.kmvp.data.repository.ApiClient
import com.woowrale.kmvp.data.repository.ApiService
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ServicesModule {

    @Provides
    @Singleton
    fun getApiServices():ApiService{
        return ApiClient.getclient().create(ApiService::class.java)
    }
}
